api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -30,25 +30,62 @@
  * Represents a manifest file that can be scanned to find data files in a table.
  */
 public interface ManifestFile {
-  Schema SCHEMA = new Schema(
-      required(500, "manifest_path", Types.StringType.get()),
-      required(501, "manifest_length", Types.LongType.get()),
-      required(502, "partition_spec_id", Types.IntegerType.get()),
-      optional(503, "added_snapshot_id", Types.LongType.get()),
-      optional(504, "added_data_files_count", Types.IntegerType.get()),
-      optional(505, "existing_data_files_count", Types.IntegerType.get()),
-      optional(506, "deleted_data_files_count", Types.IntegerType.get()),
-      optional(507, "partitions", Types.ListType.ofRequired(508, Types.StructType.of(
-          required(509, "contains_null", Types.BooleanType.get()),
-          optional(510, "lower_bound", Types.BinaryType.get()), // null if no non-null values
-          optional(511, "upper_bound", Types.BinaryType.get())
-      ))),
-      optional(512, "added_rows_count", Types.LongType.get()),
-      optional(513, "existing_rows_count", Types.LongType.get()),
-      optional(514, "deleted_rows_count", Types.LongType.get()));
+  Types.NestedField PATH = required(500, "manifest_path", Types.StringType.get());
```

Comment:
I'm going to move the schemas to a different file to avoid leaking them in the API, but I'll leave the `schema` method that returns the current unified read schema.